packages/python/plotly/_plotly_utils/utils.py

```diff
@@ -184,8 +184,13 @@ def encode_as_numpy(obj):
 
         if obj is numpy.ma.core.masked:
             return float("nan")
-        else:
-            raise NotEncodable
+        elif isinstance(obj, numpy.ndarray):
+            try:
+                return numpy.datetime_as_string(obj).tolist()
```

Comment:
this doesn't try to coerce large ints into dates does it?

Reply:
The traceback's claim that it rejects non datetime values
```
np.datetime_as_string([1, 2, 3])

Traceback (most recent call last):
  File "/media/jmmease/SSD2/anaconda/envs/plotly_dev36/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-52e13b098ed9>", line 1, in <module>
    np.datetime_as_string([1, 2, 3])
  File "<__array_function__ internals>", line 6, in datetime_as_string
TypeError: input must have type NumPy datetime
```

but we could be more conservative and check the dtype kind first

Reply:
I think that would be good, yes.

Reply:
Done in https://github.com/plotly/plotly.py/pull/3022/commits/7d5023bbfa2fd5b0f08933828aad06c42f50bb10